GeneBe.pm

```diff
@@ -0,0 +1,165 @@
+        GeneBe_ACMG_classification => "ACMG classification automatically assigned by GeneBe algorithm",
+        GeneBe_ACMG_criteria => "ACMG criteria applied by GeneBe algorithm",
+    };
+}
+
+sub run {
+    my ($self, $tva) = @_;
+
+    $self->{has_cache} = 1;
+
+    my %hg_assembly = ( #'grch37' => 'hg19',
+        'grch38' => 'hg38' );
+    my $assembly = lc($self->{config}->{assembly});
+
+    # only works on human
+    die("ERROR: GeneBe ACMG plugin works only on human data") unless $self->{config}->{species} =~ /human|homo/i;
+
+    # only work on 1 human assembly right now
+    die("ERROR: GeneBe ACMG plugin works only on human assembly hg38 ".join(' and ', keys(%hg_assembly))) unless $hg_assembly{$assembly};
+
+
+
+    # get the VF object
+    my $vf = $tva->variation_feature;
+    my $transcript = $tva->transcript;
+    my $end = $vf->{end};
+    my $start = $vf->{start};
+
+    my $ref_allele = $vf->ref_allele_string;
+    my $alt_alleles = $tva->base_variation_feature->alt_alleles;
+    my $allele_number = $tva->{allele_number};
+    my $alt_allele = @$alt_alleles[$allele_number - 1];
+
+    return {} unless defined $vf;
+
+    my $feature = $vf;
+
+    # Convert LRG to chromosome system
+    if ( $feature->coord_system_name() eq "lrg" ) {
+        $feature = $vf->transform('chromosome');
+        unless ( $feature ) {
+            warn "Region $vf->{chr}:$vf->{start}_$vf->{end} not defined in " .
+                 "chromosome coordinate system\n";
+            return {};
+        }
+    }
+    my $chr = $feature->slice->seq_region_name();
+    $chr =~ s/^chr//;
+
+    my $locus = sprintf 'chr%s-%s-%s-%s', $chr, $start, $ref_allele, $alt_allele;
+
+    # check the cache
+    if(!exists($self->{lovd_cache}->{$locus})) {
+
+        # Define the base URL of your API
+        my $base_url = 'https://api.genebe.net/cloud/api-public/v1/variant';
+
+        # Construct the URL with the variables
+        my $api_url = sprintf('%s?chr=%s&pos=%d&ref=%s&alt=%s&genome=hg38',
+                      $base_url, $chr, $start, $ref_allele, $alt_allele);
+
+        # Execute curl command to make the HTTP request
+        my $curl_command = "curl --netrc -s -A GeneBe_VEP_plugin \"$api_url\"";
+        my $curl_output = `$curl_command`;
+
+        if($? == 0) {
+
+            # Decode the JSON response
+            my $json_response = decode_json($curl_output);
+
+            # Check if 'variants' array exists and has at least one element
+            if (exists $json_response->{variants} && @{$json_response->{variants}}) {
+                # Extract data from the first element
+                my $variant_data = $json_response->{variants}[0];
+
+                # Extract required values
+                my $acmg_score = $variant_data->{acmg_score} // '';
+                my $acmg_classification = $variant_data->{acmg_classification} // '';
+                my $acmg_criteria = $variant_data->{acmg_criteria} // '';
```

Comment:
if the acmg_criteria is not defined, instead of an empty string you could use `'.'` or `'-'`.

At the moment, this is how the default output looks like when the criteria is empty:
`GeneBe_ACMG_classification=Uncertain_significance;GeneBe_ACMG_criteria=;GeneBe_ACMG_score=0;`

It would look better like this:
`GeneBe_ACMG_classification=Uncertain_significance;GeneBe_ACMG_criteria=.;GeneBe_ACMG_score=0;`

Reply:
Thanks, implemented in https://github.com/Ensembl/VEP_plugins/pull/714/commits/766dc03d8bf6c943b54c899c7d05cf50fab1c742